atc/testhelpers/match_plan_matcher.go

```diff
@@ -79,73 +79,17 @@ func (matcher *PlanMatcher) NegatedFailureMessage(actual interface{}) string {
 }
 
 func stripIDs(plan atc.Plan) (atc.Plan, []string) {
-	var ids []string
-
-	var subIDs []string
-
-	plan.ID = "<stripped>"
-
-	if plan.Aggregate != nil {
-		for i, p := range *plan.Aggregate {
-			(*plan.Aggregate)[i], subIDs = stripIDs(p)
-			ids = append(ids, subIDs...)
-		}
-	}
-
-	if plan.InParallel != nil {
-		for i, p := range plan.InParallel.Steps {
-			plan.InParallel.Steps[i], subIDs = stripIDs(p)
-			ids = append(ids, subIDs...)
-		}
-	}
-
-	if plan.Do != nil {
-		for i, p := range *plan.Do {
-			(*plan.Do)[i], subIDs = stripIDs(p)
-			ids = append(ids, subIDs...)
-		}
-	}
-
-	if plan.OnSuccess != nil {
-		plan.OnSuccess.Step, subIDs = stripIDs(plan.OnSuccess.Step)
-		ids = append(ids, subIDs...)
-
-		plan.OnSuccess.Next, subIDs = stripIDs(plan.OnSuccess.Next)
-		ids = append(ids, subIDs...)
-	}
-
-	if plan.OnFailure != nil {
-		plan.OnFailure.Step, subIDs = stripIDs(plan.OnFailure.Step)
-		ids = append(ids, subIDs...)
-
-		plan.OnFailure.Next, subIDs = stripIDs(plan.OnFailure.Next)
-		ids = append(ids, subIDs...)
-	}
-
-	if plan.Ensure != nil {
-		plan.Ensure.Step, subIDs = stripIDs(plan.Ensure.Step)
-		ids = append(ids, subIDs...)
-
-		plan.Ensure.Next, subIDs = stripIDs(plan.Ensure.Next)
-		ids = append(ids, subIDs...)
-	}
-
-	if plan.Timeout != nil {
-		plan.Timeout.Step, subIDs = stripIDs(plan.Timeout.Step)
-		ids = append(ids, subIDs...)
-	}
-
-	if plan.Try != nil {
-		plan.Try.Step, subIDs = stripIDs(plan.Try.Step)
-		ids = append(ids, subIDs...)
-	}
-
-	if plan.Get != nil {
-		if plan.Get.VersionFrom != nil {
-			planID := atc.PlanID("<stripped>")
-			plan.Get.VersionFrom = &planID
+	// Ignore errors, since our walker doesn't return an error.
+	plan.Each(func(plan *atc.Plan) {
+		plan.ID = "<stripped>"
+
+		if plan.Get != nil {
+			if plan.Get.VersionFrom != nil {
+				planID := atc.PlanID("<stripped>")
+				plan.Get.VersionFrom = &planID
+			}
 		}
-	}
+	})
 
-	return plan, ids
```

Comment:
I couldn't convince myself that this used to work. I believe the function always returned a `nil` value for `ids`.  I'm happy to fix it as part of this PR.

Reply:
Oh yeah there was definitely a bug in the test 😅  It should be testing against the IDs that it stripped. I think it'd be great if we could fix it as part of this PR, I think the only thing needed to fix it is to append the ids to a list and return that? Thanks for catching that!

Reply:
Fixing this is proving to be a little tricky. The execute command creates a new plan factory using the current timestamp to seed the factory. I am not sure how we can test this in the integration test. Any ideas?

Reply:
You're totally right, that is pretty impossible to test by checking to see if the ids stripped in the actual vs expected is the same.

Maybe we can test the plan ids a different way, by just checking to see if the ids within the actual plan are all unique? It would be basically the exact same as the `planner_test.go` here

https://github.com/concourse/concourse/blob/350e79a1385df809cbabba41040635cc3d715b43/atc/builds/planner_test.go#L670-L682

Reply:
I pushed the change.